### ISSUE TITLE
AM randomly crash on startup

### ISSUE THREAD

**Issue by gnuhead-chieb** [MuntashirAkon/AppManager#656]
- [x] I know what my device, OS and App Manager versions are
- [x] I know how to take logs
- [x] I know how to reproduce the issue which may not be specific to my device

**Crash logs**
[logcat_01-20-2022_20-57-47.txt](https://github.com/MuntashirAkon/AppManager/files/7904944/logcat_01-20-2022_20-57-47.txt)

**Comment by MuntashirAkon** [comment]
Not possible to debug without the steps to reproduce it.

**Comment by gnuhead-chieb** [comment]
I don't know what the cause is.Just tap icon of AM icon to open and crash AM with this error sometime(not all time)

**Comment by MuntashirAkon** [comment]
I have received another report via Email and another on in Telegram too:

```
java.lang.RuntimeException: Parcel android.os.Parcel@2525814: Unmarshalling unknown type code 31 at offset 612
    at android.os.Parcel.readValue(Parcel.java:3253)
    at android.os.Parcel.readSparseArrayInternal(Parcel.java:3650)
    at android.os.Parcel.readSparseArray(Parcel.java:2792)
    at android.os.Parcel.readValue(Parcel.java:3231)
    at android.os.Parcel.readArrayMapInternal(Parcel.java:3567)
    at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
    at android.os.BaseBundle.unparcel(BaseBundle.java:236)
    at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
    at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2207)
    at android.app.Activity.onRestoreInstanceState(Activity.java:1731)
    at android.app.Activity.performRestoreInstanceState(Activity.java:1684)
    at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3570)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2152)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:250)
    at android.app.ActivityThread.main(ActivityThread.java:7877)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:958)

Device Info:
App version: 2.7.0-1669-DEBUG
App version code: 395
Android build version: e8a3c
Android release version: 11
Android SDK version: 30
Android build ID: RPCS31.Q2-109-21-3-1
Device brand: motorola
Device manufacturer: motorola
Device name: minsk
Device model: moto g stylus (2021)
Device product name: minsk_t
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: root 
```

This is something to do with restoring the app state and seems completely random thus far.

**Comment by AadilGillani** [comment]
https://stackoverflow.com/questions/21342700/proguard-causing-runtimeexception-unmarshalling-unknown-type-code-in-parcelabl

**Comment by MuntashirAkon** [comment]
> https://stackoverflow.com/questions/21342700/proguard-causing-runtimeexception-unmarshalling-unknown-type-code-in-parcelabl
>
> This might be helpful ig

Not at all. Proguard/minification is completely disabled in App Manager.

**Comment by MuntashirAkon** [comment]
Could be an issue with the AppCompat library as well. Sometimes when App Manager crashes, it crashes two items (the second crash is this one). I, at first, thought this might be something to do with the new search widget but then I seem to have been getting the same crash all over the place i.e. not just the activity with this particular widget.

**Comment by MuntashirAkon** [comment]
Crash logs have now changed a bit:

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: io.github.muntashirakon.AppManager.debug, PID: 29850
    java.lang.RuntimeException: Parcel android.os.Parcel@54491c5: Unmarshalling unknown type code 6881399 at offset 1612
        at android.os.Parcel.readValue(Parcel.java:3253)
        at android.os.Parcel.readSparseArrayInternal(Parcel.java:3650)
        at android.os.Parcel.readSparseArray(Parcel.java:2792)
        at android.os.Parcel.readValue(Parcel.java:3231)
        at android.os.Parcel.readArrayMapInternal(Parcel.java:3567)
        at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
        at android.os.BaseBundle.unparcel(BaseBundle.java:236)
        at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
        at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2181)
        at android.app.Activity.onRestoreInstanceState(Activity.java:1679)
        at android.app.Activity.performRestoreInstanceState(Activity.java:1632)
        at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3473)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7664)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by MuntashirAkon** [comment]
Crash logs have changed again:

```
java.lang.RuntimeException: Parcel android.os.Parcel@fa0a5ab: Unmarshalling 
unknown type code 6881399 at offset 1612
     at android.os.Parcel.readValue(Unknown Source:48)
     at android.os.Parcel.readSparseArrayInternal(Unknown Source:6)
     at android.os.Parcel.readSparseArray(Unknown Source:13)
     at android.os.Parcel.readValue(Unknown Source:136)
     at android.os.Parcel.readArrayMapInternal(Unknown Source:6)
     at android.os.BaseBundle.initializeFromParcelLocked(Unknown Source:63)
     at android.os.BaseBundle.unparcel(Unknown Source:8)
     at android.os.Bundle.getSparseParcelableArray(Unknown Source:0)
     at com.android.internal.policy.PhoneWindow.restoreHierarchyState(Unknown 
Source:8)
     at android.app.Activity.onRestoreInstanceState(Unknown Source:14)
     at android.app.Activity.performRestoreInstanceState(Unknown Source:0)
     at android.app.Instrumentation.callActivityOnRestoreInstanceState(Unknown 
Source:0)
     at android.app.ActivityThread.handleStartActivity(Unknown Source:70)
     at 
android.app.servertransaction.TransactionExecutor.performLifecycleSequence(Unknown 
Source:124)
     at android.app.servertransaction.TransactionExecutor.cycleToPath(Unknown 
Source:10)
     at 
android.app.servertransaction.TransactionExecutor.executeLifecycleState(Unknown 
Source:25)
     at android.app.servertransaction.TransactionExecutor.execute(Unknown Source:76)
     at android.app.ActivityThread$H.handleMessage(Unknown Source:127)
     at android.os.Handler.dispatchMessage(Unknown Source:19)
     at android.os.Looper.loopOnce(Unknown Source:176)
     at android.os.Looper.loop(Unknown Source:76)
     at android.app.ActivityThread.main(Unknown Source:138)
     at java.lang.reflect.Method.invoke(Native Method)
     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Unknown 
Source:11)
     at com.android.internal.os.ZygoteInit.main(Unknown Source:309)

Device Info:
App version: 3.0.0-alpha01
App version code: 400
Android build version: 22.2.9
Android release version: 12
Android SDK version: 31
Android build ID: SKQ1.211209.001 release-keys
Device brand: Redmi
Device manufacturer: Xiaomi
Device name: alioth
Device model: M2012K11AC
Device product name: alioth
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: ru-RU
In-App Language: auto
Mode: auto
```

**Comment by gnuhead-chieb** [comment]
it seems it often happens when after installed new apps.

**Comment by MuntashirAkon** [comment]
> it seems it often happens when after installed new apps.

Yes, because this is when the activity is restarted, thus, triggering the error.

**Comment by PsyEng** [comment]
I know, I doesn't provide a crashlog or anything else, but maybe this will help to investigate the problem.
I had the same problem like the others (no apps after the update) but also lost a shortcut, which I wasn't able to restore in any way. I tried to downgrade via F-droid (which didn't work), after that I uninstaled the alpha and installed the previous version, which seems to purge all settings from AM and after this I uninstaled the current version and installed the alpha, cause I didn't see any hope to restore my provius settinga/shortcuts.

**Comment by gnuhead-chieb** [comment]
> Yes, because this is when the activity is restarted, thus, triggering the error.

May `onRestoreInstanceState` help?I don't know Android Java,so Im not sure.
https://github.com/twoyi/twoyi/commit/92860fa9b28320e8109bc7f627b601202c63df10

**Comment by MuntashirAkon** [comment]
> May `onRestoreInstanceState` help?I don't know Android Java,so Im not sure.
> [twoyi/twoyi@92860fa](https://github.com/twoyi/twoyi/commit/92860fa9b28320e8109bc7f627b601202c63df10)

Disabling old instance state breaks the paradigm.

**Comment by gnuhead-chieb** [comment]
after https://github.com/MuntashirAkon/AppManager/issues/430 fix,this error seems to be apparently decreasing

**Comment by MuntashirAkon** [comment]
It might be something to do with the AndroidX's SplashScreen library.

**Comment by kryptobolt200528** [comment]
java.lang.RuntimeException: Parcel android.os.Parcel@b258b25: Unmarshalling unknown type code [6881398](tel:6881398) at offset 644
    at android.os.Parcel.readValue([Parcel.java:3269](http://parcel.java:3269/))
    at android.os.Parcel.readSparseArrayInternal([Parcel.java:3666](http://parcel.java:3666/))
    at android.os.Parcel.readSparseArray([Parcel.java:2808](http://parcel.java:2808/))
    at android.os.Parcel.readValue([Parcel.java:3247](http://parcel.java:3247/))
    at android.os.Parcel.readArrayMapInternal([Parcel.java:3583](http://parcel.java:3583/))
    at android.os.BaseBundle.initializeFromParcelLocked([BaseBundle.java:292](http://basebundle.java:292/))
    at android.os.BaseBundle.unparcel([BaseBundle.java:236](http://basebundle.java:236/))
    at android.os.Bundle.getSparseParcelableArray([Bundle.java:1079](http://bundle.java:1079/))
    at [com.android](http://com.android/).internal.policy.PhoneWindow.restoreHierarchyState([PhoneWindow.java:2197](http://phonewindow.java:2197/))
    at [android.app](http://android.app/).Activity.onRestoreInstanceState([Activity.java:1736](http://activity.java:1736/))
    at [android.app](http://android.app/).Activity.performRestoreInstanceState([Activity.java:1689](http://activity.java:1689/))
    at [android.app](http://android.app/).Instrumentation.callActivityOnRestoreInstanceState([Instrumentation.java:1357](http://instrumentation.java:1357/))
    at [android.app](http://android.app/).ActivityThread.handleStartActivity([ActivityThread.java:3759](http://activitythread.java:3759/))
    at [android.app](http://android.app/).servertransaction.TransactionExecutor.performLifecycleSequence([TransactionExecutor.java:232](http://transactionexecutor.java:232/))
    at [android.app](http://android.app/).servertransaction.TransactionExecutor.cycleToPath([TransactionExecutor.java:212](http://transactionexecutor.java:212/))
    at [android.app](http://android.app/).servertransaction.TransactionExecutor.executeLifecycleState([TransactionExecutor.java:184](http://transactionexecutor.java:184/))
    at [android.app](http://android.app/).servertransaction.TransactionExecutor.execute([TransactionExecutor.java:102](http://transactionexecutor.java:102/))
    at [android.app](http://android.app/).ActivityThread$H.handleMessage([ActivityThread.java:2307](http://activitythread.java:2307/))
    at android.os.Handler.dispatchMessage([Handler.java:106](http://handler.java:106/))
    at android.os.Looper.loop([Looper.java:254](http://looper.java:254/))
    at [android.app](http://android.app/).ActivityThread.main([ActivityThread.java:8243](http://activitythread.java:8243/))
    at java.lang.reflect.Method.invoke(Native Method)
    at [com.android](http://com.android/).internal.os.RuntimeInit$[MethodAndArgsCaller.run](http://methodandargscaller.run/)([RuntimeInit.java:612](http://runtimeinit.java:612/))
    at [com.android](http://com.android/).internal.os.ZygoteInit.main([ZygoteInit.java:1006](http://zygoteinit.java:1006/))

Device Info:
App version: 3.0.0-alpha02-DEBUG
App version code: 401
Android build version: [1639416523335](tel:1639416523335)
Android release version: 11
Android SDK version: 30
Android build ID: RMX2063_11_C.21
Device brand: realme
Device manufacturer: realme
Device name: RMX2063L1
Device model: RMX2063
Device product name: RMX2063
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: root

**Comment by MuntashirAkon** [comment]
The crashes have decreased even further. But still receiving reports. Example:

```
java.lang.RuntimeException: Parcel android.os.Parcel@1d21e8: Unmarshalling unknown type code 6881399 at offset 1696
   at android.os.Parcel.readValue(Parcel.java:3305)
   at android.os.Parcel.readSparseArrayInternal(Parcel.java:3706)
   at android.os.Parcel.readSparseArray(Parcel.java:2844)
   at android.os.Parcel.readValue(Parcel.java:3283)
   at android.os.Parcel.readArrayMapInternal(Parcel.java:3623)
   at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
   at android.os.BaseBundle.unparcel(BaseBundle.java:236)
   at android.os.Bundle.getSparseParcelableArray(Bundle.java:1080)
   at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2198)
   at android.app.Activity.onRestoreInstanceState(Activity.java:1722)
   at android.app.Activity.performRestoreInstanceState(Activity.java:1675)
   at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1388)
   at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3748)
   at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
   at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
   at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
   at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
   at android.os.Handler.dispatchMessage(Handler.java:106)
   at android.os.Looper.loopOnce(Looper.java:201)
   at android.os.Looper.loop(Looper.java:288)
   at android.app.ActivityThread.main(ActivityThread.java:7870)
   at java.lang.reflect.Method.invoke(Native Method)
   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

Device Info:
App version: 3.0.0-alpha03
App version code: 402
Android build version: 8353555
Android release version: 12
Android SDK version: 32
Android build ID: SP2A.220505.002
Device brand: google
Device manufacturer: Google
Device name: raven
Device model: Pixel 6 Pro
Device product name: raven
Device hardware name: raven
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: root
```

**Comment by metronightly** [comment]
Yep, happened to me just few minutes ago: 
java.lang.RuntimeException: Parcel android.os.Parcel@22eded3: Unmarshalling unknown type code 6881398 at offset 644
    at android.os.Parcel.readValue(Parcel.java:3269)
        at android.os.Parcel.readSparseArrayInternal(Parcel.java:3666)
            at android.os.Parcel.readSparseArray(Parcel.java:2808)
                at android.os.Parcel.readValue(Parcel.java:3247)
                    at android.os.Parcel.readArrayMapInternal(Parcel.java:3583)
                        at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
                            at android.os.BaseBundle.unparcel(BaseBundle.java:236)
                                at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
                                    at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2197)
                                        at android.app.Activity.onRestoreInstanceState(Activity.java:1741)
                                            at android.app.Activity.performRestoreInstanceState(Activity.java:1694)
                                                at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
                                                    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3777)
                                                        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
                                                            at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
                                                                at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
                                                                    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
                                                                        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2328)
                                                                            at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                at android.os.Looper.loop(Looper.java:255)
                                                                                    at android.app.ActivityThread.main(ActivityThread.java:8212)
                                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                                            at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
                                                                                                at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
Device Info:
App version: 3.0.0-rc04
App version code: 406
Android build version: 1652239672752
Android release version: 11
Android SDK version: 30
Android build ID: RMX2027_11_C.11
Device brand: realme
Device manufacturer: realme
Device name: RMX2027
Device model: RMX2027
Device product name: RMX2027
Device hardware name: mt6768
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: no-root

**Comment by metronightly** [comment]
(Posting this comment again after finally figuring out the formatting)

Yep, same random crash on launching AM happened to me multiple times:

```
java.lang.RuntimeException: Parcel android.os.Parcel@22eded3: Unmarshalling unknown type code 6881398 at offset 644
   at android.os.Parcel.readValue(Parcel.java:3269)
   at android.os.Parcel.readSparseArrayInternal(Parcel.java:3666)
   at android.os.Parcel.readSparseArray(Parcel.java:2808)
at android.os.Parcel.readValue(Parcel.java:3247)
at android.os.Parcel.readArrayMapInternal(Parcel.java:3583)
at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
at android.os.BaseBundle.unparcel(BaseBundle.java:236)
at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2197)
at android.app.Activity.onRestoreInstanceState(Activity.java:1741)
at android.app.Activity.performRestoreInstanceState(Activity.java:1694)
at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3777)
at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2328)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loop(Looper.java:255)
at android.app.ActivityThread.main(ActivityThread.java:8212)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
Device Info:
App version: 3.0.0-rc04
App version code: 406
Android build version: 1652239672752
Android release version: 11
Android SDK version: 30
Android build ID: RMX2027_11_C.11
Device brand: realme
Device manufacturer: realme
Device name: RMX2027
Device model: RMX2027
Device product name: RMX2027
Device hardware name: mt6768
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: no-root
```



The following crash happened to me while I was switching from another app to AM via recent apps tray. This too has happened multiple times to me;

```
java.lang.ClassCastException: io.github.muntashirakon.widget.MaterialAlertView cannot be cast to java.lang.Object[]
    at androidx.collection.SimpleArrayMap.allocArrays(SimpleArrayMap.java:184)
    at androidx.collection.SimpleArrayMap.put(SimpleArrayMap.java:458)
    at androidx.transition.TransitionManager$MultiListener.onPreDraw(TransitionManager.java:281)
    at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:1093)
    at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:3575)
    at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:2340)
    at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:9057)
    at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1040)
    at android.view.Choreographer.doCallbacks(Choreographer.java:864)
    at android.view.Choreographer.doFrame(Choreographer.java:791)
    at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1025)
    at android.os.Handler.handleCallback(Handler.java:938)
    at android.os.Handler.dispatchMessage(Handler.java:99)
    at android.os.Looper.loop(Looper.java:255)
    at android.app.ActivityThread.main(ActivityThread.java:8212)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)
Device Info:
App version: 3.0.0-rc04
App version code: 406
Android build version: 1652239672752
Android release version: 11
Android SDK version: 30
Android build ID: RMX2027_11_C.11
Device brand: realme
Device manufacturer: realme
Device name: RMX2027
Device model: RMX2027
Device product name: RMX2027
Device hardware name: mt6768
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: no-root
```

**Comment by metronightly** [comment]
Another crash that happened just now.
```
java.lang.RuntimeException: Parcel android.os.Parcel@a667bf8: Unmarshalling unknown type code 6881398 at offset 644
    at android.os.Parcel.readValue(Parcel.java:3269)
    at android.os.Parcel.readSparseArrayInternal(Parcel.java:3666)
    at android.os.Parcel.readSparseArray(Parcel.java:2808)
    at android.os.Parcel.readValue(Parcel.java:3247)
    at android.os.Parcel.readArrayMapInternal(Parcel.java:3583)
    at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
    at android.os.BaseBundle.unparcel(BaseBundle.java:236)
    at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
    at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2197)
    at android.app.Activity.onRestoreInstanceState(Activity.java:1741)
    at android.app.Activity.performRestoreInstanceState(Activity.java:1694)
    at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3777)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2328)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:255)
    at android.app.ActivityThread.main(ActivityThread.java:8212)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:632)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1049)

Device Info:
App version: 3.0.0-rc04
App version code: 406
Android build version: 1652239672752
Android release version: 11
Android SDK version: 30
Android build ID: RMX2027_11_C.11
Device brand: realme
Device manufacturer: realme
Device name: RMX2027
Device model: RMX2027
Device product name: RMX2027
Device hardware name: mt6768
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en
In-App Language: en
Mode: no-root
```

**Comment by citizenserious** [comment]
java.lang.RuntimeException: Parcel android.os.Parcel@77c3b29: Unmarshalling unknown type code 6881398 at offset 644
    at android.os.Parcel.readValue(Parcel.java:3253)
    at android.os.Parcel.readSparseArrayInternal(Parcel.java:3650)
    at android.os.Parcel.readSparseArray(Parcel.java:2792)
    at android.os.Parcel.readValue(Parcel.java:3231)
    at android.os.Parcel.readArrayMapInternal(Parcel.java:3567)
    at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:292)
    at android.os.BaseBundle.unparcel(BaseBundle.java:236)
    at android.os.Bundle.getSparseParcelableArray(Bundle.java:1079)
    at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2181)
    at android.app.Activity.onRestoreInstanceState(Activity.java:1679)
    at android.app.Activity.performRestoreInstanceState(Activity.java:1632)
    at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1356)
    at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3473)
    at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
    at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
    at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
    at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
    at android.os.Handler.dispatchMessage(Handler.java:106)
    at android.os.Looper.loop(Looper.java:223)
    at android.app.ActivityThread.main(ActivityThread.java:7664)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

Device Info:
App version: 3.0.0
App version code: 410
Android build version: eng.user.20220730.123708
Android release version: 11
Android SDK version: 30
Android build ID: lineage_starlte-userdebug 11 RQ3A.211001.001 eng.user.20220730.123708 test-keys
Device brand: samsung
Device manufacturer: samsung
Device name: starlte
Device model: SM-G960F
Device product name: starltexx
Device hardware name: samsungexynos9810
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: en-US
In-App Language: auto
Mode: auto

**Comment by MuntashirAkon** [comment]
I've received more than 15 crash reports in the past month. So, I think it's time to fix this issue once and for all.

And I've found a way to reproduce it easily:

1. Open App Manager
2. Open Android Settings app and go to the app info page for App Manager there
3. Revoke/grant file management permission
4. Go back to App Manager
5. App Manager crashes with the error message shared above.

Now, the best way to fix this issue is to find exactly which build has caused it. It was first reported in build#1669. So, I shall start from that build and go backwards to find out.

**Comment by MuntashirAkon** [comment]
It seems the issue was introduced in build#1636 (784a49a5e7982e2601b3c1db6293375933bf3f1c)

**Comment by MuntashirAkon** [comment]
Fixed in 7498b4cabea2458496bd8fda516d4d8f41324662

**Comment by George-Seven** [comment]
Thank you for the fix, works great!
